march_hardware/src/IMotionCube.cpp

```diff
@@ -110,409 +91,182 @@ void IMotionCube::writeInitialSettings(uint8 ecatCycleTime)
+  }
 
-  if (this->mosiByteOffsets.count(IMCObjectName::TargetTorque) != 1)
+  if (target_torque >= MAX_TARGET_TORQUE)
   {
-    ROS_WARN("TargetTorque not defined in PDO mapping, so can't do actuateTorque");
-    return;
+    throw error::HardwareException(error::ErrorType::TARGET_TORQUE_EXCEEDS_MAX_TORQUE,
+                                   "Target torque of %d exceeds max torque of %d", target_torque, MAX_TARGET_TORQUE);
   }
-  uint8 targetTorqueLocation = this->mosiByteOffsets[IMCObjectName::TargetTorque];
+
+  bit16 target_torque_struct;
+  target_torque_struct.i = target_torque;
+
+  uint8_t target_torque_location = this->mosi_byte_offsets_.at(IMCObjectName::TargetTorque);
 
   ROS_DEBUG("Trying to actuate slave %d, soem location %d with target torque %d", this->slaveIndex,
-            targetTorqueLocation, targetTorqueStruct.i);
-  set_output_bit16(this->slaveIndex, targetTorqueLocation, targetTorqueStruct);
+            target_torque_location, target_torque_struct.i);
+  set_output_bit16(this->slaveIndex, target_torque_location, target_torque_struct);
 }
 
 float IMotionCube::getAngleRad()
 {
-  ROS_ASSERT_MSG(this->misoByteOffsets.count(IMCObjectName::ActualPosition) == 1, "ActualPosition not defined in PDO "
-                                                                                  "mapping, so can't get angle");
-  return this->encoder.getAngleRad(this->misoByteOffsets[IMCObjectName::ActualPosition]);
+  return this->encoder_.getAngleRad(this->miso_byte_offsets_.at(IMCObjectName::ActualPosition));
 }
 
 float IMotionCube::getTorque()
 {
-  ROS_ASSERT_MSG(this->misoByteOffsets.count(IMCObjectName::ActualTorque) == 1, "ActualTorque not defined in PDO "
-                                                                                "mapping, so can't get torque");
-  union bit16 return_byte = get_input_bit16(this->slaveIndex, this->misoByteOffsets[IMCObjectName::ActualTorque]);
+  bit16 return_byte = get_input_bit16(this->slaveIndex, this->miso_byte_offsets_.at(IMCObjectName::ActualTorque));
   ROS_DEBUG("Actual Torque read: %d", return_byte.i);
   return return_byte.i;
 }
 
 int IMotionCube::getAngleIU()
 {
-  ROS_ASSERT_MSG(this->misoByteOffsets.count(IMCObjectName::ActualPosition) == 1, "ActualPosition not defined in PDO "
-                                                                                  "mapping, so can't get angle");
-  return this->encoder.getAngleIU(this->misoByteOffsets[IMCObjectName::ActualPosition]);
+  return this->encoder_.getAngleIU(this->miso_byte_offsets_.at(IMCObjectName::ActualPosition));
 }
 
-uint16 IMotionCube::getStatusWord()
+uint16_t IMotionCube::getStatusWord()
 {
-  ROS_ASSERT_MSG(this->misoByteOffsets.count(IMCObjectName::StatusWord) == 1, "StatusWord not defined in PDO "
-                                                                              "mapping, so can't get status word");
-  return get_input_bit16(this->slaveIndex, this->misoByteOffsets[IMCObjectName::StatusWord]).ui;
```

Comment:
See commit https://github.com/project-march/hardware-interface/pull/202/commits/26d64d63dfaee741bdc53ba444e3e5780d453236

> This is replaced by using the `.at()` method on the maps. This method throws an exception when the key does not exist.

The asserts were not warnings and would cause the program to abort abruptly, which is not nice to do and cannot be caught. On the other hand, this is something you can not recover from, since this should never happen.